src/app/core/auth/auth.interceptor.ts

```diff
@@ -3,26 +3,55 @@ import {
     HttpEvent, HttpInterceptor, HttpHandler, HttpRequest
 } from '@angular/common/http';
 import { isPlatformBrowser } from '@angular/common';
-
 import { Observable } from 'rxjs';
+import { ConfigService } from '@gsrs-core/config/config.service';
 
 @Injectable()
 export class AuthInterceptor implements HttpInterceptor {
 
-    constructor(@Inject(PLATFORM_ID) private platformId: any) {
+    constructor(
+        @Inject(PLATFORM_ID) private platformId: any,
+        public configService: ConfigService) {
     }
 
     intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
-
         if (isPlatformBrowser(this.platformId)) {
+
+            // Reading from config file, and adding additional header
+            if (this.configService.configData) {
+                if (this.configService.configData.authenticateAs.apiUsername !== null) {

```

Comment:
Need to check if if `this.configService.configData.authenticateAs` is null/present first. Lots of config files won't have it, and that's okay